framework/Distributions.py

```diff
@@ -1704,6 +1708,133 @@ def rvs(self):
 
 DistributionsCollection.addSub(Categorical.getInputSpecification())
 
+class MarkovCategorical(Categorical):
+  """
+    Class for the Markov categorical distribution based on "Markov Model"
+    Note: this distribution can have only numerical (float) outcome; in the future we might want to include also the possibility to give symbolic outcome
+  """
+
+  @classmethod
+  def getInputSpecification(cls):
+    """
+      Method to get a reference to a class that specifies the input data for
+      class cls.
+      @ In, cls, the class for which we are retrieving the specification
+      @ Out, inputSpecification, InputData.ParameterInput, class to use for
+        specifying input of cls.
+    """
+    inputSpecification = InputData.parameterInputFactory(cls.__name__, ordered=True, baseNode=None)
+
+    StatePartInput = InputData.parameterInputFactory("state", contentType=InputData.StringType)
+    StatePartInput.addParam("outcome", InputData.FloatType, True)
+    StatePartInput.addParam("index", InputData.IntegerType, True)
+    TransitionInput = InputData.parameterInputFactory("transition", contentType=InputData.StringType)
```

Comment:
Instead of a string, you should you the ListTypeFloat so you should not need to cast the values manually.